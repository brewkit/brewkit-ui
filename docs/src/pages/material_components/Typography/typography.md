# Typography

## Extends [Typography](https://v4.mui.com/components/typography/#typography)
