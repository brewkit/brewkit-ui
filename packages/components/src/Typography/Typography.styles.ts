import { Theme } from '@material-ui/core/styles/createTheme';
import { makeStyles } from '@material-ui/core/styles';

const useStyles = makeStyles<Theme>(
  (theme: Theme) => ({
    primary: { color: theme.palette.primary.main },
    secondary: { color: theme.palette.secondary.main },
    success: { color: theme.palette.success.main },
    error: { color: theme.palette.error.main },
    warning: { color: theme.palette.warning.main },
    text: { color: theme.palette.text.primary },
  }),
  { name: 'Typography' }
);

export default useStyles;
