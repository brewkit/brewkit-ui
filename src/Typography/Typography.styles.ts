import { createStyles, makeStyles } from '@material-ui/core/styles';

const useStyles = makeStyles(
    (theme) =>
        createStyles({
            primary: { color: theme.palette.primary.main },
            secondary: { color: theme.palette.secondary.main },
            success: { color: theme.palette.success.main },
            error: { color: theme.palette.error.main },
            warning: { color: theme.palette.warning.main },
            text: { color: theme.palette.text.primary },
        }),
    { name: 'Typography' },
);

export default useStyles;
