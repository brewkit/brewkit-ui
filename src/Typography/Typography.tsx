import React from 'react';
import clsx from 'clsx';
import { merge } from 'lodash';
import {
    Typography as MUITypography,
    TypographyProps as MUITypographyProps,
} from '@material-ui/core';
import withoutKeys from '../utils/withoutKeys';
import useStyles from './Typography.styles';

export interface BkTypographyProps extends Omit<MUITypographyProps, 'color'> {
    color?:
        | 'primary'
        | 'secondary'
        | 'error'
        | 'success'
        | 'warning'
        | 'textPrimary'
        | 'textSecondary';
}

const Typography = React.forwardRef(
    (props: BkTypographyProps, ref: React.Ref<HTMLElement>) => {
        const {
            children,
            className,
            classes: userClasses = {},
            color = 'textPrimary',
            variant = 'body1',
            ...otherProps
        } = props;
        const typographyClasses = useStyles();
        const classes = merge(typographyClasses, userClasses);
        const muiClasses = withoutKeys(classes, [
            'error',
            'primary',
            'secondary',
            'success',
            'textPrimary',
            'textSecondary',
            'warning',
        ]);

        return (
            <MUITypography
                className={clsx(className, classes?.[color])}
                classes={muiClasses}
                color={undefined} // We manually resolve colors color
                ref={ref}
                variant={variant}
                {...otherProps}>
                {children}
            </MUITypography>
        );
    },
);

Typography.displayName = 'BkTypography';

export default Typography;
