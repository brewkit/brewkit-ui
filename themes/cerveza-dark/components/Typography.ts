

import { color } from '../../cerveza/variables';


const Typography = {


    overrides: {

        MuiTypography: {

            subtitle1: {
                color: color.blue.light,
            },

            subtitle2: {
                color: color.blue.light,
            },

            button: {
                color: color.blue.main,
            },

            body1: {
                color: color.blueGray.light,
            },

            body2: {
                color: color.blueGray.light,
            },

            caption : {
                color: color.blueGray.light,
            }

        },

    },


};


export default Typography;
