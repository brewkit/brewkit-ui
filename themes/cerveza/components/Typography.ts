import { color } from '../variables';


const Typography = {


    typography: {
        fontSize: 16,
        h1: {
            fontWeight: 'bold',
            fontSize: '5rem',
            lineHeight: 1.2,
        },
        h2: {
            fontWeight: 'bold',
            fontSize: '3rem',
            lineHeight: 1.25,
        },
        h3: {
            fontWeight: 'bold',
            fontSize: '2rem',
            lineHeight: 1.5,
        },
        h4: {
            fontWeight: 'bold',
            fontSize: '1.5rem',
            lineHeight: 1.33,
        },
        h5: {
            fontWeight: 'bold',
            fontSize: '1.25rem',
            lineHeight: 1.2,
        },
        h6: {},
        body1: {
            lineHeight: 1.25,
        },
        body2: {
            lineHeight: 1.25,
        },
        subtitle1: {
            fontWeight: 'bold',
            fontSize: '1.5rem',
            lineHeight: 1.33,
        },
        subtitle2: {
            fontWeight: 'bold',
            fontSize: '1.25rem',
            lineHeight: 1.2,
        },
        button: {
            fontWeight: 'bold',
            fontSize: '1.25rem',
            lineHeight: 1.2,
        },
        caption: {
            fontWeight: 'bold',
            fontSize: '1.25rem',
            lineHeight: 1.2,
        },
        overline: {
            fontWeight: 'bold',
            fontSize: '1.25rem',
            lineHeight: 1.2,
        },
    },


    overrides: {

        MuiTypography: {

            subtitle1: {
                color: color.blue.dark,
            },

            subtitle2: {
                color: color.blue.dark,
            },

            button: {
                color: color.blue.main,
            },

            body1: {
                color: color.blueGray.dark,
            },

            body2: {
                color: color.blueGray.dark,
            },

        },

    },


};


export default Typography;
