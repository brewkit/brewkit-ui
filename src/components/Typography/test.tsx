import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Typography from './index';


it('renders correctly', () => {
    const tree = renderer
        .create(<Typography>Foo</Typography>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});


it('renders children when passed in', () => {
    const wrapper = shallow((
        <Typography>
            <span>foo</span>
        </Typography>
    ));
    expect(wrapper.contains(<span>foo</span>)).toEqual(true);
});


it('accepts custom classes', () => {
    const wrapper = shallow((
        <Typography className="customClass">Lorem ipsum dolor sit amet.</Typography>
    ));
    expect(wrapper.hasClass('customClass')).toEqual(true);
    expect(wrapper.hasClass('brew-Typography')).toEqual(true);
});


it('"variant" defaults to "inherit" and applies appropriate class', () => {
    const wrapper = shallow((
        <Typography>Lorem ipsum dolor sit amet.</Typography>
    ));
    expect(wrapper.hasClass('brew-Typography--variant-inherit')).toEqual(true);
});


it('"color" defaults to "initial" and applies appropriate class', () => {
    const wrapper = shallow((
        <Typography>Lorem ipsum dolor sit amet.</Typography>
    ));
    expect(wrapper.hasClass('brew-Typography--color-initial')).toEqual(true);
});


it('"display" defaults to "initial" and applies appropriate class', () => {
    const wrapper = shallow((
        <Typography>Lorem ipsum dolor sit amet.</Typography>
    ));
    expect(wrapper.hasClass('brew-Typography--display-initial')).toEqual(true);
});


it('"align" defaults to "inherit" and applies appropriate class', () => {
    const wrapper = shallow((
        <Typography>Lorem ipsum dolor sit amet.</Typography>
    ));
    expect(wrapper.hasClass('brew-Typography--align-inherit')).toEqual(true);
});
