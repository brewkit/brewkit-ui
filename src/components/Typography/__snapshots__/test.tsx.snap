// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<span
  className="brew-Typography brew-Typography--variant-inherit brew-Typography--color-initial brew-Typography--align-inherit brew-Typography--display-initial"
>
  Foo
</span>
`;
